### PR TITLE
Add Workflow logs and fix installed version workflow bug

### LINKED ISSUE #4425: sideBySide Experimental Feature Causing GetVariant(14) 0x80070490 Error - Winget v1.8.1133-preview
### Brief description of your issue

The sideBySide Experimental Features causes variant issue when trying to install an application Winget as the source and setting "sideBySide: true" under experimental features.

I also submitted to the Feedback Hub with all of my Winget logs and diagnostic data recorded. Find it here:

[https://aka.ms/AAq7fke](https://aka.ms/AAq7fkel)

The video attached is a side-by-side recording of my Terminal and Winget "settings.json" so you can see exactly how sideBySide was enabled, install failed, sideBySide disabled, install succeeds.

https://github.com/microsoft/winget-cli/assets/67825086/ddb77b54-d21b-4011-a39b-54ad311a70f9

### Steps to reproduce

Steps:
1. Enable sideBySide
2. ```winget install Google.Chrome.Beta -s winget```
3. ```Unexpected Error - GetVariant(14) 0x80070490 : Element not found.```
4. Disable sideBySide
5. Open new shell
6. ```winget install Google.Chrome.Beta -s winget```
7. Google Chrome Beta installs as expected

Also tried with ```winget install Google.Chrome.Dev -s winget```

### Expected behavior

C:\>```winget install Google.Chrome.Beta -s winget```
Found Google Chrome Beta [Google.Chrome.Beta] Version 125.0.6422.14
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://dl.google.com/dl/chrome/install/beta/googlechromebetastandaloneenterprise64.msi
  ██████████████████████████████   116 MB /  116 MB
Successfully verified installer hash
Starting package install...
Successfully installed

C:\>

### Actual behavior

C:\>```winget install Google.Chrome.Beta -s winget```
An unexpected error occurred while executing the command:
GetVariant(14)
0x80070490 : Element not found.

C:\>

### Environment

```shell
My current Winget settings.json:
{
    "$schema": "https://aka.ms/winget-settings.schema.json",
    "experimentalFeatures": {
        "experimentalCmd": true,
        "experimentalArg": true,
	    "dependencies": true,
	    "directMSI": true,
        "resume": true,
        "configuration03": true,
        "configureSelfElevate": true,
        "sideBySide": false
    },
    "visual": {
        "progressBar": "rainbow",
        "anonymizeDisplayedPaths": false
    },
    "source": {
        "autoUpdateIntervalInMinutes": 60
    },
    "installBehavior": {
        "disableInstallNotes": false,
        "skipDependencies": false,
        "includePreviewVersions": true,
        "portablePackageUserRoot": "C:\\Portable-Apps",
        "portablePackageMachineRoot": "C:\\Portable-Apps",
        "allowPrerelease": true,
        "allowExperimental": true,
        "preferences": {
            "locale": ["en-US"],
            "architectures": ["x64"],
            "installerTypes": ["msix", "msi", "exe", "msstore", "wix", "nullsoft", "zip", "inno", "burn",  "portable"]
        }
    },
    "uninstallBehavior": {
        "purgePortablePackage": true
    },
    "network": {
       "downloader": "wininet"
    },
    "telemetry": {
        "disable": false    
    },
    "logging": {
        "level": "verbose",
        "channels": ["all"]
    },
    "interactivity": {
        "disable": false
    }
}

Winget --info --verbose:
Windows Package Manager (Preview) v1.8.1133-preview
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.26200.5001
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.23.1133.0

Winget Directories
-----------------------------------------------------------------------------------------------------------------------------------------
Logs                               C:\Users\david\AppData\Local\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutp…
User Settings                      C:\Users\david\AppData\Local\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\settings…
Portable Links Directory (User)    C:\Users\david\AppData\Local\Microsoft\WinGet\Links
Portable Links Directory (Machine) C:\Program Files\WinGet\Links
Portable Package Root (User)       C:\Portable-Apps
Portable Package Root              C:\Portable-Apps
Portable Package Root (x86)        C:\Portable-Apps
Installer Downloads                C:\Users\david\Downloads

Links
---------------------------------------------------------------------------
Privacy Statement   https://aka.ms/winget-privacy
License Agreement   https://aka.ms/winget-license
Third Party Notices https://aka.ms/winget-3rdPartyNotice
Homepage            https://aka.ms/winget
Windows Store Terms https://www.microsoft.com/en-us/storedocs/terms-of-sale

Admin Setting                             State
--------------------------------------------------
LocalManifestFiles                        Disabled
BypassCertificatePinningForMicrosoftStore Disabled
InstallerHashOverride                     Disabled
LocalArchiveMalwareScanOverride           Disabled
ProxyCommandLineOptions                   Disabled
DefaultProxy                              Disabled

C:\>
```

### PR DESCRIPTION
**Pull request by JohnMcPMS**
Fixes #4425 

## Change
This adds a `Workflow` channel (disabled by default) that has the entry to tasks and the `Get`/`Add`/`Contains` calls logged to it.  This enables a view into a flow that can be collected by a third party.

It also fixes a behavior change with the side-by-side code path for selecting the installed version that would lead to no data item being inserted, when the previous behavior was that a `nullptr` would be inserted when no version is installed.

## Validation
Manual validation of the workflow output, including using the windbg command to see the function name (for function pointer tasks).
Manual validation of at least one case of the `GetVariant(14)` issue being fixed.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-cli/pull/4432)

### CODE REVIEW

File: src/AppInstallerCLICore/ExecutionContext.cpp [hunk 1]
```diff
@@ -487,6 +487,24 @@ namespace AppInstaller::CLI::Execution
...
+            break;
+        case EnumBasedVariantMapAction::Contains:
+            AICLI_LOG(Workflow, Info, << "Checking data item: " << data);
+            break;
+        case EnumBasedVariantMapAction::Get:
+            AICLI_LOG(Workflow, Info, << "Getting data item: " << data);
+            break;
+        }
+
+        UNREFERENCED_PARAMETER(map);
```
Comment (yao-msft) [inline src/AppInstallerCLICore/ExecutionContext.cpp:505]: Do we plan to use this variable in the future?

File: src/AppInstallerRepositoryCore/CompositeSource.cpp [hunk 2]
```diff
@@ -1365,6 +1365,8 @@ namespace AppInstaller::Repository
         }
     }
 
+    using namespace anon;
```
Comment (yao-msft) [inline src/AppInstallerRepositoryCore/CompositeSource.cpp:1368]: curious why do we need to change the empty namespace?

**Review by yao-msft — APPROVED** [review]

File: src/AppInstallerRepositoryCore/CompositeSource.cpp (hunk 2 above)
Comment (JohnMcPMS) [reply src/AppInstallerRepositoryCore/CompositeSource.cpp:1368]: I've learned that anonymous namespace does not output type information into the PDB in a way that debuggers can find currently, making it much harder to debug.  With a named namespace, the derived type information is found and so the debugger can show the member fields of a derived type (vtable -> type mapping).  One can manually do all of this with enough effort, and maybe eventually the debuggers will support it directly, but at this time I see simply naming the namespace as the easiest answer.

File: src/AppInstallerCLICore/ExecutionContext.cpp (hunk 1 above)
Comment (JohnMcPMS) [reply src/AppInstallerCLICore/ExecutionContext.cpp:505]: I had considered a much more complex runtime checking with tasks actually having information on which data they produced/consumed, but realized that a runtime check is only as good as the tests and so provided very little value.  I didn't feel like removing the parameter at that point.

**Pull request merged**